### PR TITLE
Send key up events for the modifier keys on exit

### LINKED ISSUE #15: Key up events for modifier keys are not sent on exit
The virtual keyboard sends the keydown and keyup events for `Shift`, `Ctrl`, and `Alt` separately. 

This makes it possible for a user to send the keydown event for a modifier and then exit before sending the keyup. 

On my computer (KDE neon 5.22, X server 1.20.9, kernel 5.4.0-77-generic), this results in all input events being handled as if those modifier keys were pressed. I can only fix it by manually sending the keyup event for the correct modifier key.

I have created a pull request that solves this by dispatching the keyup events for Shift_R, Control_R, and Alt_R in VncActivity.onStop.

### PR DESCRIPTION
**Pull request by thechosenreader**
This resolves #15

### CODE REVIEW

**Comment by gujjwal00** [comment]
Although this approach solves the issue, it seems slightly broad to me. Now we will always send key-up events, even if user has not used the virtual keys. Also, we will do this in every `onStop()` call, irrespective of whether our activity is finishing or simply going in background.  

Another minor issue is that virtual key state can become obsolete. Suppose you press 'Ctrl' but then receive a call. Our `onStop()` will be called and we will send key-up events. But when call is finished and you go back to AVNC, 'Ctrl' would still be pressed.


What do you think about simply toggling-off the meta key buttons from  `onStop()`? If any of them is pressed, it will automatically trigger the required key-up events. We can create a  `releaseMetaKeys()` function in 
 `VirtualKeys` and call it from  `onStop()`.

**Comment by thechosenreader** [comment]
This seems like a much better idea :)

I've done as you suggested. I had to create id's for the meta key buttons in the virtual_keys layout and then called their `toggle()` method in `releaseMetaKeys`.

That said, I can't help but feel like we're only bandaging the problem, instead of really fixing it. I may be way off base here - I have literally less than an hour of experience with this code-base - but might there be a better solution, one that also takes care of the similar problems with the mouse buttons? Perhaps there is something in the VNC specification that will help address this in a better way? I'm not really sure, so I'll defer that to your expertise.

In the mean time, I've committed the above changes.

File: app/src/main/java/com/gaurav/avnc/ui/vnc/VirtualKeys.kt [hunk 1]
```diff
@@ -48,6 +48,13 @@ class VirtualKeys(activity: VncActivity) {
         }
     }
 
+    fun releaseMetaKeys() {
+        val binding = stub.binding as VirtualKeysBinding
+        binding.shiftBtn.toggle()
+        binding.ctrlBtn.toggle()
+        binding.altBtn.toggle()
```
Comment (gujjwal00) [inline app/src/main/java/com/gaurav/avnc/ui/vnc/VirtualKeys.kt:55]: Please note that `stub.binding` will be `null` until virtual keys are inflated, so we have to use safe cast `as?` here.
Also,  `toggle()` will *invert* the button state, but we want to always release the buttons.

Comment (gujjwal00) [inline app/src/main/java/com/gaurav/avnc/ui/vnc/VirtualKeys.kt:55]:
```suggestion
        val binding = stub.binding as? VirtualKeysBinding
        binding?.apply {
            shiftBtn.isChecked = false
            ctrlBtn.isChecked = false
            altBtn.isChecked = false
        }
```

**Comment by gujjwal00** [comment]
> That said, I can't help but feel like we're only bandaging the problem, instead of really fixing it. I may be way off base here - I have literally less than an hour of experience with this code-base - but might there be a better solution, one that also takes care of the similar problems with the mouse buttons? Perhaps there is something in the VNC specification that will help address this in a better way? I'm not really sure, so I'll defer that to your expertise.

This is something where we will always end up bandaging the problem. Whatever we do, network/connection can always go down after we send a key-down event and we will have the same issue. It's more in the hands of VNC servers here. They can be a little more defensive, and release all keys when a client disconnects.

For our side, I can think of two alternative approaches:
1. Release meta-keys from `cleanup()` in `Messenger`, which will be called when we are disconnecting. And as we are disconnecting, we won't have to worry about virtual keys. 
2. Delay the key-down events until we are sure about key-up. For example, if 'Ctrl' is followed by 'a', we delay 'Ctrl' key-down until we have received 'a', and then release 'Ctrl' automatically. But it would probably be very complicated to sync everything up because for some keys we don't want this behavior (e.g. 'Tab'), and most character have separate key-down/key-up events.

**Pull request merged**